crystal_diffusion/samplers/variance_sampler.py

```diff
@@ -21,15 +25,52 @@ class NoiseParameters:
     sigma_min: float = 0.005
     sigma_max: float = 0.5
 
+    # Default value comes from "Generative Modeling by Estimating Gradients of the Data Distribution"
+    corrector_step_epsilon: float = 2e-5
+
 
 class ExplodingVarianceSampler:
     """Exploding Variance Sampler.
 
-    This class is responsible for creating the all the quantities
-    needed for noise generation.
+    This class is responsible for creating all the quantities needed
+    for noise generation for training and sampling.
 
     This implementation will use "exponential diffusion" as discussed in
-    the paper "Torsional Diffusion for Molecular Conformer Generation".
+    the papers (no one paper presents everything clearly)
+        - [1] "Torsional Diffusion for Molecular Conformer Generation".
+        - [2] "SCORE-BASED GENERATIVE MODELING THROUGH STOCHASTIC DIFFERENTIAL EQUATIONS"
+        - [3] "Generative Modeling by Estimating Gradients of the Data Distribution"
+
+    The following quantities are defined:
+        - total number of times steps, N
+
+        - time steps:
+            t in [delta, 1], on an even discretized grid, t_i for i = 1, ..., N.
+            We avoid t = 0 because sigma(t) is poorly defined there. See Appendix C of  [2]
+
+        - sigma and sigma^2:
+            standard deviation, following the "exploding variance scheme",
+                sigma(t) = sigma_min^(1-t) sigma_max^t.
+                sigma_i for i = 1, ..., N.
+
+        - g and g^2:
+            g is the diffusion coefficient that appears in the stochastic differential equation (SDE).
+                g^2(t) = d sigma^2(t)/ dt for the exploding variance scheme. This becomes discretized as
+                g^2_i = sigma^2_{i} - sigma^2_{i-1} for i = 1, .., N.
+
+                --> The papers never clearly state what to do for sigma_{i=0}.
+                    We CHOOSE sigma_{i=0} = sigma_min = sigma(t=0)
+
+        - eps and sqrt_2_eps:
+            This is for Langevin dynamics within a corrector step. Following [3], we define
+
+                    eps_i = 0.5 epsilon_step * sigma^2_i / sigma^2_1 for i = 0, ..., N-1.
+
+                --> Careful! eps_0 is needed for the corrector steps.
+
+
+
+
```

Comment:
bunch of empty lines that could be removed

Reply:
done :+1:.